src/Components/Aspire.Pomelo.EntityFrameworkCore.MySql/AspireEFMySqlExtensions.cs

```diff
@@ -0,0 +1,165 @@
+        }
+
+        if (builder.Configuration.GetConnectionString(connectionName) is string connectionString)
+        {
+            settings.ConnectionString = connectionString;
+        }
+
+        configureSettings?.Invoke(settings);
+
+        if (settings.DbContextPooling)
+        {
+            builder.Services.AddDbContextPool<TContext>(ConfigureDbContext);
+        }
+        else
+        {
+            builder.Services.AddDbContext<TContext>(ConfigureDbContext);
+        }
+
+        if (settings.HealthChecks)
+        {
+            // calling MapHealthChecks is the responsibility of the app, not Component
+            builder.TryAddHealthCheck(
+                name: typeof(TContext).Name,
+                static hcBuilder => hcBuilder.AddDbContextCheck<TContext>());
+        }
+
+        if (settings.Tracing)
+        {
+            builder.Services.AddOpenTelemetry()
+                .WithTracing(tracerProviderBuilder =>
+                {
+                    // add tracing from the underlying MySqlConnector ADO.NET library
+                    tracerProviderBuilder.AddSource("MySqlConnector");
+                });
+        }
+
+        if (settings.Metrics)
+        {
+            builder.Services.AddOpenTelemetry()
+                .WithMetrics(meterProviderBuilder =>
+                {
+                    // Currently EF provides only Event Counters:
+                    // https://learn.microsoft.com/ef/core/logging-events-diagnostics/event-counters?tabs=windows#counters-and-their-meaning
+                    meterProviderBuilder.AddEventCountersInstrumentation(eventCountersInstrumentationOptions =>
+                    {
+                        // The magic strings come from:
+                        // https://github.com/dotnet/efcore/blob/a1cd4f45aa18314bc91d2b9ea1f71a3b7d5bf636/src/EFCore/Infrastructure/EntityFrameworkEventSource.cs#L45
+                        eventCountersInstrumentationOptions.AddEventSources("Microsoft.EntityFrameworkCore");
+                    });
+
+                    // add metrics from the underlying MySqlConnector ADO.NET library
+                    meterProviderBuilder.AddMeter("MySqlConnector");
+                });
+        }
+
+        void ConfigureDbContext(IServiceProvider serviceProvider, DbContextOptionsBuilder dbContextOptionsBuilder)
+        {
+            // use the legacy method of setting the ILoggerFactory because Pomelo EF Core doesn't use MySqlDataSource
+            if (serviceProvider.GetService<ILoggerFactory>() is { } loggerFactory)
+            {
+                MySqlConnectorLogManager.Provider = new MicrosoftExtensionsLoggingLoggerProvider(loggerFactory);
+            }
+
+            // We don't register logger factory, because there is no need to: https://learn.microsoft.com/dotnet/api/microsoft.entityframeworkcore.dbcontextoptionsbuilder.useloggerfactory?view=efcore-7.0#remarks
```

Comment:
This comment is kind of weird given that directly above it we are explicitly registering a logger factory.

Reply:
Was copy/pasted; will delete.